lib/facter/self_service.rb

```diff
@@ -25,11 +24,14 @@
   end
 
   chunk(:S0004) do
-    next unless PuppetSelfService.primary?
     # Is PE and has clienttools covers pe-psql and compilers
     # Check for service status that is not green, potentially need a better way of doing this, or perhaps calling the api directly for each service
-    result = Facter::Core::Execution.execute("#{puppet_bin} infrastructure status")
-    if result.include?('Unknown') || result.include?('Unreachable')
+
+    puppetendpoint = 'https://localhost:8140/status/v1/services'
+
+    puppetendpointcall = PuppetSelfService.nethttp_puppet_api(puppetendpoint)
+
+    if puppetendpointcall.include?('Unknown') || puppetendpointcall.include?('Unreachable')
```

Comment:
needs to fail on  the following status's

error if any service reports an error
starting if any service reports that it is starting, and no service reports an error or that it is stopping
stopping if any service reports that it is stopping and no service reports an error
unknown if any service reports an unknown state and no services report an error